src/cursor/cur_layered.c

```diff
@@ -128,8 +128,9 @@ __clayered_enter(WT_CURSOR_LAYERED *clayered, bool reset, bool update)
          *   - an update operation with a ingest cursor, or
          *   - a read operation and the cursor is open for reading.
          */
-        if ((update && clayered->ingest_cursor != NULL) ||
-          (!update && F_ISSET(clayered, WT_CLAYERED_OPEN_READ)))
+        if (clayered->stable_cursor != NULL &&
```

Comment:
The current `open_cursors` implementation always opens cursors on all constituents, so these checks might be overly complex (I think your change here bypasses the latter test that you left untouched). Is my understanding right?

Could you update the comment?